crates/symbolicator/src/services/download/mod.rs

```diff
@@ -374,6 +383,13 @@ impl Iterator for SourceLocationIter<'_> {
     }
 }
 
+/// Computes a download timeout based on a content length. If the content length is
+/// `None` (i.e., unknown), assume 2GB. The minimum timeout is 1s.
+fn content_length_timeout(content_length: Option<u32>, streaming_timeout: Duration) -> Duration {
```

Comment:
Given the `max_download_timeout` already existing at the top level.  Isn't the intentions and interactions here clearer to express in the code by returning an `Option<Duration>` here if there is no content length?  The maximum download is enforced anyway.

And I don't mean this entirely literally that changing the return type is the best solution.  Maybe the best solution is to not even call this function, i don't know yet as I haven't tried this out.  But the point i'm trying to make is that the interactions and intentions are probably clearer if no streaming timeout is applied if no content-length is known.